src/components/ratings-and-reviews-components/review-and-rating-post-form/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { SyntheticEvent, useEffect } from 'react';
+import { Review } from '@services/generated-api/data-contracts';
+import api from '@services/api';
+import { useFormAndValidation } from '@hooks/use-form-and-validation';
+import RatingInput from '@components/ratings-and-reviews-components/rating-input';
+import Button from '@components/Button';
+import styles from './review-and-rating-post-form.module.scss';
+
+const dateOptions = { day: '2-digit', month: '2-digit', year: '2-digit' };
+
+interface IReviewAndRatingPostForm {
+	review: Review | null;
+	productId: number;
+	onAddReview: (review: Review) => void;
+	onUpdateReview: (review: Review) => void;
+}
+
+const ReviewAndRatingPostForm: React.FC<IReviewAndRatingPostForm> = ({
+	review,
+	productId,
+	onAddReview,
+	onUpdateReview,
+}) => {
+	const { values, setValues, handleChange } = useFormAndValidation({
+		text: review?.text || '',
+		score: review?.score || 0,
+	});
+
+	useEffect(() => {
+		review && setValues({ text: review?.text, score: review?.score });
+	}, [review, productId, setValues]);
+
+	const handleReviewSubmit = (e: SyntheticEvent) => {
+		e.preventDefault();
+		// TODO: Add error processing in catch block
+		if (review?.score && values.text && values.text !== review?.text) {
+			api
+				.reviewsUpdate(productId, review.id, { text: values.text as string })
+				.then((res) => onUpdateReview(res))
+				.catch((err) => console.log(err));
+		}
+	};
+
+	const handleRatingChange = (rating: number) => {
```

Comment:
Старайся по чаще декомпозировать код, а то его местами из-за нарастающей логики + большого уровня вложенности становится сложно читать)

```suggestion
	const handleRatingChange = (rating: number) => {
	const handleReviewUpdate = (res: any) => {
		onUpdateReview(res);
		return res;
	};

	const handleReviewCreate = (res: any) => {
		onAddReview(res);
		return res;
	};

	if (review?.score === rating) return;

	const updateReview = review && review.score > 0
		? api.reviewsUpdate(productId, review.id, { score: rating }).then(handleReviewUpdate)
		: api.reviewsCreate(productId, { score: rating }).then(handleReviewCreate);

	updateReview
		.then(() => setValues({ ...values, score: rating }))
		.catch((err) => console.log(err));
};```

Надеюсь ничего не сломал)